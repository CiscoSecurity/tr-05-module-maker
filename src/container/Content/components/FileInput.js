import React from "react";
import "./FileInput.scss"

const FileInput = ({value, onChange, ...rest}) => (
    <div className='img-wrapper'>
        <input type="file" onChange={onChange} className="img-upload" {...rest}/>
        <div className="drag">
            {Boolean(value) && (
                <div className="img">
                    <div>Selected file:</div>
                    <img src={value} alt="" width="50"/>
                </div>
            )}
        </div>
    </div>
);

export default FileInput;