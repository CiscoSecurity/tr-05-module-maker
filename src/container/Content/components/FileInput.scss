input.img-upload {
  color: #f9f9f9;
  width: 98%;
  border: 1px dashed #000;
  vertical-align: middle;

  &:hover {
    border: 1px dashed #67a5c5;
  }
}

.img-wrapper {
  .drag {
    align-items: end;
    min-width: 80px;
    margin-top: -80px;
    margin-bottom: 47px;
    align-content: center;
    vertical-align: middle;
    padding-left: 10px;
    font-size: 12px;
  }
}
