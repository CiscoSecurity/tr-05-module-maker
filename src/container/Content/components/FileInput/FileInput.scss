@import "src/globals/constants/colors";

.imgWrapper {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;

  .imgUpload {
    border: none;
    color: transparent;
    width: 90px;
  }

  .closeIcon {
    fill: $gray-color;
    width: 15px;
    height: 15px;
    margin: 4px;
    vertical-align: top;

    &:hover {
      fill: $hover-color;
      cursor: pointer;
    }
  }
}
