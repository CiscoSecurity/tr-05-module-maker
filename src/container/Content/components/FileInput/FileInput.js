import React, { useCallback } from "react";
import "./FileInput.scss"
import Icons from "globals/icons/sprite.svg";
import { deleteLogo, onFileLoaded } from "../../additionalInputsActions";
import { connect } from "react-redux";
import { useDropzone } from 'react-dropzone'


const FileInput = ({ value, deleteLogo, onFileLoaded }) => {
    const onDrop = useCallback((acceptedFiles) => {
        const file = acceptedFiles[0]
        if (file) {
            const reader = new FileReader()
            reader.onload = () => {
                const dataURL = reader.result
                onFileLoaded(dataURL)
            }
            reader.readAsDataURL(file)
        }

    }, [onFileLoaded])
    const {getRootProps, getInputProps, isDragActive} = useDropzone({
        onDrop, multiple: false, accept: "image/*"
    })

    return (
        <div className="dropzone-wrapper">
            <div {...getRootProps({className: `dropzone${isDragActive ? '-active' : ''}`})}>
                <input {...getInputProps()} />
                <p>Drag 'n' drop your logo here, or click to select logo</p>
            </div>
            {Boolean(value) &&
            (<div>
                <img src={value} alt="" height="60" className="img"/>
                <svg className="closeIcon" onClick={deleteLogo}>
                    <use xlinkHref={`${Icons}#icon-small-x-close`}/>
                </svg>
            </div>)
            }
        </div>
    )
}


const mapDispatchToProps = {
    deleteLogo,
    onFileLoaded
}

export default connect(null, mapDispatchToProps)(FileInput);
