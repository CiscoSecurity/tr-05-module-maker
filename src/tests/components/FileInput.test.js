import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import FileInput from "../../container/Content/components/FileInput/FileInput";
import React from "react";
import configureStore from 'redux-mock-store';
import * as mocks from "../mocks/mockPayloads"
import toJson from "enzyme-to-json";


const mockStore = configureStore([]);


describe('FileInput', () => {
    const handleLoadLocalFile = jest.fn();

    it('should render FileInput with image preview', () => {
            const store = mockStore({
                logo: mocks.LOAD_FILE_MOCK,
            });

        const wrapper = mount(
            <Provider store={store}>
                <FileInput  value={mocks.LOAD_FILE_MOCK}
                            onChange={handleLoadLocalFile}/>
            </Provider>);

        expect(wrapper.find('img')).toHaveLength(1);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should render FileInput without image preview', () => {
        const store = mockStore({
            logo: mocks.LOAD_FILE_MOCK,
        });

        const wrapper = mount(
            <Provider store={store}>
                <FileInput  value={''}
                            onChange={handleLoadLocalFile}/>
            </Provider>);

        expect(wrapper.find('img')).toHaveLength(0);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
});
