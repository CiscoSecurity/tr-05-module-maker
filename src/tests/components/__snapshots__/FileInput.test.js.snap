// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput should render FileInput with image preview 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(FileInput)
    onChange={[MockFunction]}
    value="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
  >
    <FileInput
      deleteLogo={[Function]}
      onChange={[MockFunction]}
      onFileLoaded={[Function]}
      value="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
    >
      <div
        className="dropzone-wrapper"
      >
        <div
          className="dropzone"
          onBlur={[Function]}
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <input
            accept="image/*"
            autoComplete="off"
            multiple={false}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            tabIndex={-1}
            type="file"
          />
          <p>
            Drag 'n' drop your logo here, or click to select logo
          </p>
        </div>
        <div>
          <img
            alt=""
            className="img"
            height="60"
            src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
          />
          <svg
            className="closeIcon"
            onClick={[Function]}
          >
            <use
              xlinkHref="sprite.svg#icon-small-x-close"
            />
          </svg>
        </div>
      </div>
    </FileInput>
  </Connect(FileInput)>
</Provider>
`;

exports[`FileInput should render FileInput without image preview 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(FileInput)
    onChange={[MockFunction]}
    value=""
  >
    <FileInput
      deleteLogo={[Function]}
      onChange={[MockFunction]}
      onFileLoaded={[Function]}
      value=""
    >
      <div
        className="dropzone-wrapper"
      >
        <div
          className="dropzone"
          onBlur={[Function]}
          onClick={[Function]}
          onDragEnter={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <input
            accept="image/*"
            autoComplete="off"
            multiple={false}
            onChange={[Function]}
            onClick={[Function]}
            style={
              Object {
                "display": "none",
              }
            }
            tabIndex={-1}
            type="file"
          />
          <p>
            Drag 'n' drop your logo here, or click to select logo
          </p>
        </div>
      </div>
    </FileInput>
  </Connect(FileInput)>
</Provider>
`;
